sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignature.java

```diff
@@ -113,12 +117,148 @@ static Builder builder() {
     Method targetMethod();
   }
 
-  /** A type of optional parameter of the {@link DoFn.ProcessElement} method. */
-  public enum Parameter {
-    BOUNDED_WINDOW,
-    INPUT_PROVIDER,
-    OUTPUT_RECEIVER,
-    RESTRICTION_TRACKER
+  /** A descriptor for an optional parameter of the {@link DoFn.ProcessElement} method. */
```

Comment:
I think it's actually only ProcessElement and OnTimer. E.g. probably not StartUp or StartBundle, as a bundle can include elements from multiple keys/windows and timers/state are scoped to those. And definitely not FinishBundle or TearDown.